resources/qml/delegates/ImageMessage.qml

```diff
@@ -31,11 +31,40 @@ Item {
         fillMode: Image.PreserveAspectFit
 
         MouseArea {
+            id: mouseArea
             enabled: model.data.type == MtxEvent.ImageMessage && img.status == Image.Ready
+            hoverEnabled: true
             anchors.fill: parent
             onClicked: TimelineManager.openImageOverlay(model.data.url, model.data.id)
         }
 
-    }
+        Item {
+            id: overlay
+        
+            anchors.fill: parent
+            visible: mouseArea.containsMouse
+            
+            Rectangle {
+                id: container
+                width: parent.width
+                implicitHeight: imgcaption.implicitHeight
+                anchors.bottom: overlay.bottom 
+                color: "black"
+                opacity: 0.75
+
+                Text {
+                    id: imgcaption
 
+                    anchors.fill: parent
+                    elide: Text.ElideMiddle
+                    horizontalAlignment: Text.AlignHCenter
+                    verticalAlignment: Text.AlignVCenter
+                    // See this MSC: https://github.com/matrix-org/matrix-doc/pull/2530
+                    text: model.data.filename ? model.data.filename : model.data.body
+                    font.pointSize: 11
```

Comment:
You probably should not hardcode the font size. You probably want to use a multiple of the UserSettings.fontSize instead, so it scales with the settings.

Reply:
You're right. Commenting out the fontSize, it seems to make no difference. I'm not too sure what/how the default `font.pointSize` is set to, would it automatically use the `UserSettings.fontSize`?